sycl/test/backend/cuda/interop_get_native.cpp

```diff
@@ -0,0 +1,90 @@
+// REQUIRES: cuda
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -I%opencl_include_dir -I%cuda_toolkit_include -o %t.out -lcuda -lsycl
+// RUN: env SYCL_DEVICE_TYPE=GPU %t.out
+
+//==---------- interop_get_native.cpp - SYCL cuda get_native tests ---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <cuda.h>
+#include <iostream>
+
+using namespace cl::sycl;
+
+void check(bool condition, const char *conditionString, const char *filename,
+           const long line) noexcept {
+  if (!condition) {
+    std::cerr << "CHECK failed in " << filename << "#" << line << " "
+              << conditionString << "\n";
+    std::abort();
+  }
+}
+
+#define CHECK(CONDITION) check(CONDITION, #CONDITION, __FILE__, __LINE__)
+
+bool isCudaDevice(const device &dev) {
+  const platform platform = dev.get_info<info::device::platform>();
+  const std::string platformVersion =
+      platform.get_info<info::platform::version>();
+  // If using PI_CUDA, don't accept a non-CUDA device
+  return platformVersion.find("CUDA") != std::string::npos;
+}
+
+class cuda_device_selector : public device_selector {
+public:
+  int operator()(const device &dev) const {
+    return isCudaDevice(dev) ? 1 : -1;
+  }
+};
+
+int main() {
```

Comment:
I think the unittests primarily test the PI API. Since this tests uses SYCL I think it makes more sense to have it here. Additionally, part of the LIT test is to check that it actually compiles correctly, which I think is very relevant here given we specialize the `interop` template for the relevant backend.